### PR TITLE
Prac 05 feedback

### PR DESCRIPTION
**Pull request by a-rosa**
@nora-zyy
@AdriantoHartono
@JevanChandra

### CODE REVIEW

File: prac_05/word_occurences.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+Word Occurences
+estimate: 30 mins
+Actual: 40 mins
+"""
+
+text = input("Text: ").lower()
+words = text.split()
+word_occurences = {}
+# initialize the longest width to the width to the first word for comparison
+width = len(words[0])
```
Comment (JevanChandra) [inline prac_05/word_occurences.py:11]: Width is a bit confusing so I suggest changing the variable name to longest_word_length to make it more clear

File: prac_05/emails.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    for detail in email_details.items():
+        print(f"{detail[0]} ({detail[1]})")
+
+def get_name(email):
+    """Extract name from email"""
+    # get the front part of the email before the @ and join the names that are in between the dots
+    name = " ".join(email.split("@")[0].split(".")).title()
+    return name
+
+main()
```
Comment (JevanChandra) [inline prac_05/emails.py:30]: Great code! Very short and simple!

File: prac_05/wimbledon.py
Comment (nora-zyy) [inline prac_05/wimbledon.py:1]: The description is very clear and there are step-by-step instructions.

File: prac_05/emails.py (hunk 2 above)
Comment (nora-zyy) [reply prac_05/emails.py:30]: The coding is smooth and very easy to understand.

File: prac_05/wimbledon.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+def get_winner_occurrences(data, winners):
+    """Get winner name and how many they have won"""
+    winner_name = data.split(",")[2]
+    if winner_name in winners:
+        winners[winner_name] += 1
+    else:
+        winners[winner_name] = 1
+
+
+main()
```
Comment (JevanChandra) [inline prac_05/wimbledon.py:53]: Great code! 👍

File: prac_05/state_names.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        state_code in CODE_TO_NAME
+        print(state_code, "is", CODE_TO_NAME[state_code])
+    except:
+        print("Invalid short state")
+    state_code = input("Enter short state: ").upper()
+
+print()
+
+for state in CODE_TO_NAME:
+    print(f"{state:3} is {CODE_TO_NAME[state]}")
```
Comment (JevanChandra) [inline prac_05/state_names.py:24]: Good job!

File: prac_05/hex_colours.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+          "Indigo": "#4b0082", "Lavender": "#e6e6fa", "Olive": "808000", "Purple": "#a020f0", "SkyBlue": "#87ceeb"}
+print(COLORS)
+
+picked_color = input("Enter a colour: ").upper()
+while picked_color != "":
+    if picked_color in COLORS:
+        print(COLORS[picked_color])
+    else:
+        print("Invalid color")
+    picked_color = input("Enter a colour: ").upper()
```
Comment (JevanChandra) [inline prac_05/hex_colours.py]: Nice code! Very simple

File: prac_05/emails.py (hunk 2 above)
Comment (AdriantoHartono) [reply prac_05/emails.py:30]: Good code and simple, easy to understand 👌

File: prac_05/hex_colours.py
Comment (AdriantoHartono) [inline prac_05/hex_colours.py:1]: Good code, and it can display as expected 👍

File: prac_05/state_names.py
Comment (AdriantoHartono) [inline prac_05/state_names.py:1]: The code can work properly, good 👍

File: prac_05/wimbledon.py
Comment (AdriantoHartono) [inline prac_05/wimbledon.py:1]: The code has been written nicely with using a lot of functions 👍

File: prac_05/word_occurences.py
Comment (AdriantoHartono) [inline prac_05/word_occurences.py:1]: The code has work properly and can show the result as expected 👌

**Pull request merged**